ote_cli/ote_cli/utils/hpo.py

```diff
@@ -0,0 +1,694 @@
+        raise ValueError("Dataset is not defined.")
+
+    impl_class = get_dataset_class(task_type)
+    dataset = impl_class(
+        train_subset={
+            "ann_file": dataset_paths.get("train_ann_file", None),
+            "data_root": dataset_paths.get("train_data_root", None),
+        },
+        val_subset={
+            "ann_file": dataset_paths.get("val_ann_file", None),
+            "data_root": dataset_paths.get("val_data_root", None),
+        },
+    )
+
+    train_env = TaskEnvironment(
+        model=model,
+        hyper_parameters=hyper_parameters,
+        label_schema=generate_label_schema(dataset, task_type),
+        model_template=model_template,
+    )
+
+    hyper_parameters = train_env.get_hyper_parameters()
+
+    # set epoch
+    if task_type == TaskType.CLASSIFICATION:
+        (hyper_parameters.learning_parameters.max_num_epochs) = hp_config["iterations"]
+    elif task_type in (TaskType.DETECTION, TaskType.SEGMENTATION):
+        hyper_parameters.learning_parameters.num_iters = hp_config["iterations"]
+
+    # set hyper-parameters and print them
+    HpoManager.set_hyperparameter(hyper_parameters, hp_config["params"])
+    print(f"hyper parameter of current trial : {hp_config['params']}")
+
+    train_env.set_hyper_parameters(hyper_parameters)
+    train_env.model_template.hpo = {
+        "hp_config": hp_config,
+        "metric": hp_config["metric"],
+    }
+
+    impl_class = get_impl_class(train_env.model_template.entrypoints.base)
+    task = impl_class(task_environment=train_env)
+
+    # make environment equivalent to lr finder when hpo classification
+    if task_type == TaskType.CLASSIFICATION:
+        task.set_lr_finder_enable_flag(False)
+
+    dataset = HpoDataset(dataset, hp_config)
+    if train_env.model:
+        train_env.model.train_dataset = dataset
+        train_env.model.confiugration.configurable_parameters = hyper_parameters
+
+    output_model = ModelEntity(
+        dataset,
+        train_env.get_model_configuration(),
+        model_status=ModelStatus.NOT_READY,
+    )
+
+    # make callback to report score to hpopt every epoch
+    train_param = TrainParameters(
+        False, HpoCallback(hp_config, hp_config["metric"], task), None
+    )
+    train_param.train_on_empty_model = None
+
+    task.train(dataset=dataset, output_model=output_model, train_parameters=train_param)
+
+    hpopt.finalize_trial(hp_config)
+
+
+def exec_hpo_trainer(arg_file_name, alloc_gpus):
+    """Execute new process to train model for ASHA's trial"""
+
+    gpu_ids = ",".join(str(val) for val in alloc_gpus)
+    trainer_file_name = osp.abspath(__file__)
+    hpo_env = os.environ.copy()
+    if torch.cuda.device_count() > 0:
+        hpo_env["CUDA_VISIBLE_DEVICES"] = f"{gpu_ids}"
+    hpo_env["PYTHONPATH"] = f"{osp.dirname(__file__)}/../../:" + hpo_env.get(
+        "PYTHONPATH", ""
+    )
+
+    subprocess.run(
+        ["python", f"{trainer_file_name}", f"{arg_file_name}"],
+        shell=False,
```

Comment:
```suggestion
```

Reply:
Since there is a security issue when "shell=True", "shell=False" is explicitly written to solve this issue. (required by SWQRC)